cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/NativeDependencyResolverImpl.java

```diff
@@ -0,0 +1,312 @@
+        }
+    }
+
+    @Override
+    public List<Path> nativeDependencies(PackageID packageID) {
+        List<Path> modulePlatformLibs = new ArrayList<>();
+        // copy platform libs for all modules(imported modules as well)
+        addPlatformLibs(packageID, modulePlatformLibs);
+
+        if (isProjectModule(packageID)) {
+            return modulePlatformLibs;
+        } else if (isPathDependency(packageID)) {
+            addLibsFromBaloDependency(packageID, modulePlatformLibs);
+        } else if (isModuleInDistribution(packageID)) {
+            addLibsFromDistribution(packageID, modulePlatformLibs);
+        } else {
+            addLibsFromHomeBaloCache(packageID, modulePlatformLibs);
+        }
+        return modulePlatformLibs;
+    }
+
+    @Override
+    public List<Path> nativeDependenciesForTests(PackageID packageID) {
+        List<Path> testPlatformLibs = new ArrayList<>();
+        List<Library> libraries = manifest.getPlatform().libraries;
+        if (libraries != null) {
+            for (Library library : libraries) {
+                if ((library.getModules() == null ||
+                        Arrays.asList(library.getModules()).contains(packageID.name.value)) &&
+                        (library.getScope() != null && library.getScope().equalsIgnoreCase("testOnly"))) {
+                    String libFilePath = library.getPath();
+                    if (libFilePath == null) {
+                        continue;
+                    }
+                    Path nativeFile = sourceRootPath.resolve(Paths.get(libFilePath));
+                    testPlatformLibs.add(nativeFile);
+                }
+            }
+        }
+        return testPlatformLibs;
+    }
+
+    private boolean isModuleInDistribution(PackageID packageID) {
+        return getTomlFilePath(packageID).exists();
+    }
+
+    private File getTomlFilePath(PackageID packageID) {
+        String version = BLANG_PKG_DEFAULT_VERSION;
+        if (!packageID.version.value.equals("")) {
+            version = packageID.version.value;
+        }
+
+        return Paths.get(balHomePath, DIST_BIR_CACHE_DIR_NAME, packageID.orgName.value,
+                packageID.name.value, version, "Ballerina.toml").toFile();
+    }
+
+    private boolean isPathDependency(PackageID packageID) {
+        return buildContext.getImportPathDependency(packageID).isPresent();
+    }
+
+    private boolean isProjectModule(PackageID packageID) {
+        return manifest.getProject().getOrgName().equals(packageID.orgName.value) &&
+                ProjectDirs.isModuleExist(sourceRootPath, packageID.name.value);
+    }
+
+    private void addLibsFromHomeBaloCache(PackageID packageID, List<Path> modulePlatformLibs) {
+        for (String platform : supportedPlatforms) {
+            Path baloPath = buildContext.getBaloFromHomeCache(packageID, platform);
+            if (baloPath != null && Files.exists(baloPath)) {
```

Comment:
```suggestion
            if (baloPath != null && baloPath.toFile().exists()) {
```
According to SonarLint Java 8's `Files.exists()` method has noticibly poor performance in JDK8, So Shall we use `toFile().exists()`?